### DISCUSSION TITLE
How should show rules work for string literals?

### DISCUSSION THREAD

**Discussion by TS60** [typst/typst#4255] [Q&A] [answered]
When a show rule replaces a string that was already replaced by an other show rule, the behaviour seems to become a bit limited.
I'm trying to find out what is the limitation.
In the following example, we put a `|` between `s` and `f`:
```typst
#show "sf": "s|f"
aaasfaa
```
This works:
![img1](https://github.com/typst/typst/assets/152509161/02f504cd-fec4-417d-a542-8a96701be34d)

Now, it seems that matching the newly created `|` will also work with an additional rule:

```typst
#show "sf": "s|f"
#show "s|": "s?"
aaasfaa
```
![img2](https://github.com/typst/typst/assets/152509161/9d4a3c9d-7935-446b-adb1-9bd2835b3b1b)

But when we make the newly created show rule a little bit longer, it seems to not work anymore:

```typst
#show "sf": "s|f"
#show "as|": "s?"
aaasfaa
```
![img3](https://github.com/typst/typst/assets/152509161/188268d6-595c-4fba-b404-8e6115603646)

Am I doing something wrong or is this a known limitation?

**Comment by laurmaedje** [comment, marked as answer]
See https://github.com/typst/typst/issues/86

**Reply by TS60** [reply]
Thanks. So I assume there should not be any limitation and as long as it stays in one `Text` node it should work.
I can confirm that the following works as expected:

```typst
#show "sf": "sfff"
#show "sfff": "gg"
aaasfaa
```
